### ISSUE TITLE
Decision Tree 모형 CODE

### ISSUE THREAD

**Issue by yangwonsuck** [yangwonsuck/python_code#3]
#기본 라이브러리 불러오기
import pandas as pd
import numpy as np

# Breast Cancer 데이터셋 가져오기(출처: UCI ML Repository)
uci_path='https://archive.ics.uci.edu/ml/machine-learning-databases/breast-cancer-wisconsin/breast-cancer-wisconsin.data'
df=pd.read_csv(uci_path,header=None)

# 열 이름 지정
df.columns=['id','clump','cell_size','cell_shape','adhesion','epithlial','bare_nuclei','chromatin','normal_nucleoli','mitoses','class']

# IPython 디스플레이 설정 -출력할 열의 개수 한도 늘리기
pd.set_option('display.max_columns',15)

print(df.head())
print('\n')
#데이터 자료형 확인
print(df.info())
print('\n')
#데이터 통계 요약 정보 확인
print(df.describe())

# bare_nuclei 열의 자료형 변경(문자열 ->숫자)
print(df['bare_nuclei'].unique())
print('\n')

df['bare_nuclei'].replace('?',np.nan,inplace=True)
df.dropna(subset=['bare_nuclei'],axis=0,inplace=True)
df['bare_nuclei']=df['bare_nuclei'].astype('int')

print(df.describe())

#속성(변수) 선택
X=df[['clump','cell_size','cell_shape','adhesion','epithlial','bare_nuclei','chromatin','normal_nucleoli','mitoses']]
y=df['class']

# 설명 변수 데이터 정규화
from sklearn import preprocessing
X=preprocessing.StandardScaler().fit(X).transform(X)

#train data와 test data로 구분(7:3 비율)
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=10)

print('train data 개수: ',X_train.shape)
print('test data 개수: ',X_test.shape)

# sklearn 라이브러리에서 Decision Tree 분류 모형 가져오기
from sklearn import tree

#모형 객체 생성(criterion='entropy' 적용)
tree_model=tree.DecisionTreeClassifier(criterion='entropy',max_depth=5)

#train_data를 가지고 모형 학습
tree_model.fit(X_train,y_train)

# test data를 가지고 y_hat 예측 (분류)
y_hat=tree_model.predict(X_test)

print(y_hat[0:10])
print(y_test.values[0:10])

# 모형 성능 평가 - Confusion Matrix 계산
from sklearn import metrics
tree_matrix=metrics.confusion_matrix(y_test,y_hat)
print(tree_matrix)
print('\n')

# 모형 성능 평가 - 평가 지표 계산
tree_report=metrics.classification_report(y_test,y_hat)
print(tree_report)
